### PR TITLE
Worklist #21774: Fix Windows.browseAssets() bugs

### PR DESCRIPTION
**Pull request by Triplelexx**
Added support for list of filters using formatting consistent with other implementations, e.g: (*.png *.jpg)

**Test Plan:**
Window.browseAssets allows the selection of files from ATP.
Suitable files should first be uploaded for testing. If there are .png and .jpg files available the following script example could be used for testing.

`var asset = Window.browseAssets("Select ATP File", "/", "TestFiles(*.pNG *.jpg)");`

The first parameter is to set the window title, second is for specifying a start directory, the third is for filtering.

The extension check should be case insensitive, as with the example above.
The filter list of extensions are checked within the parentheses (  ) characters.

If filters are omitted all files should be visible, the filter parameter can also just be a single extension such as "*.png".

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9132/app_12989_PR_99ad51189c66ed5e6c2f425c55d39be9953df464.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9132/HighFidelity-Beta-PR12989-99ad51189c66ed5e6c2f425c55d39be9953df464.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9132/HighFidelity-Beta-Interface-PR12989-99ad51189c66ed5e6c2f425c55d39be9953df464.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9158/app_12989_PR_bd72350287c31a12edc65094e05febcf2343f132.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9159/app_12989_PR_bd6b6c94232d7308842df7bddbff68a9b5e7b1b6.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9159/HighFidelity-Beta-PR12989-bd6b6c94232d7308842df7bddbff68a9b5e7b1b6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9159/HighFidelity-Beta-Interface-PR12989-bd6b6c94232d7308842df7bddbff68a9b5e7b1b6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9159/HighFidelity-Beta-PR12989-bd6b6c94232d7308842df7bddbff68a9b5e7b1b6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9159/HighFidelity-Beta-Interface-PR12989-bd6b6c94232d7308842df7bddbff68a9b5e7b1b6.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9620/app_12989_PR_bd6b6c94232d7308842df7bddbff68a9b5e7b1b6.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9620/HighFidelity-Beta-PR12989-bd6b6c94232d7308842df7bddbff68a9b5e7b1b6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9620/HighFidelity-Beta-Interface-PR12989-bd6b6c94232d7308842df7bddbff68a9b5e7b1b6.dmg

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
ATP and filters look good and work as expected.

File: interface/resources/qml/dialogs/assetDialog/AssetDialogContent.qml [hunk 1]
```diff
@@ -19,14 +19,14 @@ import "../fileDialog"
...
+    property alias filter: selectionType.filtersString
     property int options  // Not used.
 
     property bool selectDirectory: false
 
     // Not implemented.
-    //property bool saveDialog: false;
-    //property bool multiSelect: false;
+    // property bool saveDialog: false;
+    // property bool multiSelect: false;
```
Comment (druiz17) [inline interface/resources/qml/dialogs/assetDialog/AssetDialogContent.qml]: If the commented code not being used anymore...please remove it.

File: interface/resources/qml/dialogs/assetDialog/AssetDialogContent.qml [hunk 2]
```diff
@@ -84,7 +84,7 @@ Item {
                 size: 28
                 width: height
                 enabled: destination !== ""
-                //onClicked: d.navigateHome();
+                // onClicked: d.navigateHome();
```
Comment (druiz17) [inline interface/resources/qml/dialogs/assetDialog/AssetDialogContent.qml]: If the commented code not being used anymore...please remove it.

Comment (Triplelexx) [reply interface/resources/qml/dialogs/assetDialog/AssetDialogContent.qml]: Double checking, should push commit tomorrow.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9665/app_12989_PR_a061e84ad7028d59dc7a8db72c070df6a9fba999.apk

**Comment by MiladNazeri** [comment]
@sethalves and @druiz17 is this good for approval now?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9665/HighFidelity-Beta-PR12989-a061e84ad7028d59dc7a8db72c070df6a9fba999.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9665/HighFidelity-Beta-Interface-PR12989-a061e84ad7028d59dc7a8db72c070df6a9fba999.dmg

**Comment by marko8904** [comment]
build this please

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9682/app_12989_PR_a061e84ad7028d59dc7a8db72c070df6a9fba999.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9682/HighFidelity-Beta-PR12989-a061e84ad7028d59dc7a8db72c070df6a9fba999.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9682/HighFidelity-Beta-Interface-PR12989-a061e84ad7028d59dc7a8db72c070df6a9fba999.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9682/HighFidelity-Beta-PR12989-a061e84ad7028d59dc7a8db72c070df6a9fba999.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9682/HighFidelity-Beta-Interface-PR12989-a061e84ad7028d59dc7a8db72c070df6a9fba999.exe

**Pull request merged**